### PR TITLE
Respect required CI checks

### PR DESCRIPTION
**Pull request by SeanMcTex**
## Summary

This PR is based on [#10](https://github.com/SeanMcTex/MonitorLizard/pull/10) by @lukelabonte (Luke LaBonte), rebased cleanly onto main and extended with a few additional fixes. All four of Luke's commits are preserved with his authorship intact.

### What Luke's work does

- Fetches required check metadata from GitHub's GraphQL API (`isRequired`, `branchProtectionRule`).
- Treats missing required contexts as pending, so a required check that hasn't reported yet doesn't show as green.
- Introduces a `notStarted` build status (gray `play.slash` icon) for PRs where required CI hasn't kicked off yet — previously these showed a misleading green checkmark.
- Keeps non-required checks out of the main merge-blocking status.
- Shows approval-gated non-blocking checks as a secondary summary row ("Non-blocking: 1 waiting for approval").
- Unifies the single-PR fetch path to use the same GraphQL query as the batch path, removing the `GHPRViewResponse` struct.
- Adds tests for required-check handling, missing required contexts, and the non-blocking summary.

### What was added on top

- **Replace name-based approval gate detection with GitHub's `WAITING` status.** The original used name heuristics ("approve"/"approval" in check names) to identify approval gates. In GitHub Actions, `CheckRun.status == WAITING` is the authoritative signal — set exclusively when a job is blocked on environment protection approval. The heuristic functions (`isApprovalGate`, `approvalGateName`, `parentWorkflowNameForApprovalGate`) are removed; `WAITING` status drives all approval gate logic. Test fixtures updated to use `WAITING` CheckRuns instead of approval-named StatusContexts.
- **Deterministic required context ordering.** `Array(Set(...))` now has `.sorted()` to give stable ordering across calls.
- **Simpler `ForEach` in non-blocking summary.** Replaced `ForEach(Array(summary.segments.enumerated()), id: \.element.id)` with `ForEach(summary.segments)` using `Identifiable` directly.

## Test plan

- [ ] Build and run on a repo that uses required CI checks — PR status should reflect required checks only
- [ ] Open a PR where required CI hasn't started yet — should show gray `play.slash`, not green checkmark
- [ ] Verify non-blocking approval-gated checks appear in the secondary summary row
- [ ] Run test suite: `xcodebuild -project MonitorLizard/MonitorLizard.xcodeproj -scheme MonitorLizard -destination 'platform=macOS' -configuration Debug CODE_SIGNING_ALLOWED=NO test`

### CODE REVIEW

File: MonitorLizard/Services/GitHubService.swift [hunk 1]
```diff
@@ -563,11 +681,106 @@ class GitHubService: ObservableObject {
         return .success
     }
 
-    private func parseStatusChecks(from checks: [GHPRDetailResponse.StatusCheck]?) -> [StatusCheck] {
+    private func hasActiveNonApprovalWork(
```
Comment (lukelabonte) [inline MonitorLizard/Services/GitHubService.swift:684]: I think this now treats every `WAITING` check as approval-related work.

That might be right for GitHub Actions environment approvals, but the code doesn’t check that the waiting check is actually an approval gate. If GitHub or another provider returns a non-approval check in `WAITING`, we could show the PR as `Not started` even though CI is active.

Could we either scope this more tightly, or add a test that proves a non-approval `WAITING` check doesn’t get hidden?

File: MonitorLizard/Services/GitHubService.swift [hunk 2]
```diff
@@ -563,11 +681,106 @@ class GitHubService: ObservableObject {
...
+            }
+            return requiredContextSet.contains(checkName)
+        }
+
+        if !requiredChecks.isEmpty {
+            return requiredChecks
+        }
+
+        let hasRequiredMetadata = requiredContextSet != nil || checks.contains { $0.isRequired != nil }
+        return hasRequiredMetadata ? [] : checks
```
Comment (lukelabonte) [inline MonitorLizard/Services/GitHubService.swift:754]: I think this drops the fallback for approval gates that come through as legacy `StatusContext` entries.

The old path handled contexts like `ci/circleci: dead_code_cleanup/approve_dead_code_cleanup` by treating them as waiting for approval. This now only works if we get a `CheckRun` with `status: WAITING`.

Are we sure external providers won’t still send approval gates as `StatusContext` + `PENDING`? If not, I think we should keep the `WAITING` path but leave the old name-based fallback for legacy contexts.

**Review by lukelabonte — COMMENTED** [review]
I left two comments on the `WAITING` handling. I think using `WAITING` is the right direction, but I’m not sure we want to make it the only approval-gate signal or treat every `WAITING` check as approval-related.

File: MonitorLizard/Services/GitHubService.swift (hunk 2 above)
Comment (SeanMcTex) [reply MonitorLizard/Services/GitHubService.swift:754]: Ok, took another swing at it. I'm not 100% happy with it, as we're still inferring approval steps based on the `jobName`, but it may be the best we can do for now. I'd be glad for your thoughts: 35a7641af0075e3157405ee8e258f7a1ed0c628c

File: MonitorLizard/Services/GitHubService.swift [hunk 3]
```diff
@@ -563,11 +681,126 @@ class GitHubService: ObservableObject {
...
+            guard let checkName = check.name ?? check.context else {
+                return false
+            }
+
+            let isRequired = check.isRequired ?? requiredContextSet.map { $0.contains(checkName) }
+
+            // Non-required WAITING checks are approval gates or approval-parent workflows — skip them.
+            // Required WAITING checks are genuine active CI work (e.g. a queued check) and must not
+            // be hidden; they fall through so hasActiveNonApprovalWork correctly returns true.
+            if check.status?.uppercased() == "WAITING", isRequired == false {
```
Comment (lukelabonte) [inline MonitorLizard/Services/GitHubService.swift]: I think this still skips too much.

This now preserves required `WAITING` checks, but any non-required `WAITING` check still returns `false` before we check whether it actually looks like an approval gate. So if a required context is missing and a non-required CI check named something like `build` is `WAITING`, we can still show the PR as `Not started` even though CI is active.

I think the shape should be closer to:

```swift
let isWaitingApprovalGate =
    check.__typename == "CheckRun"
    && check.status?.uppercased() == "WAITING"
    && isRequired == false
    && looksLikeApprovalGate(checkName)

if isWaitingApprovalGate || isLegacyApprovalContext {
    return false
}
```

And then a regression test for:

```text
required contexts:
- ci/example: required_jobs_met
- version_health / assessment

present checks:
- version_health / assessment, required, success
- build, not required, WAITING

expected overall status: pending, not notStarted
```

Could we scope the `WAITING` skip that way? I think that keeps the approval-gate behavior without hiding active non-required CI work.

**Review by lukelabonte — CHANGES_REQUESTED** [review]
I think this is close, but I'd like to see this last edge case tightened up before merging.

I left a suggested shape and test case inline.

**Comment by SeanMcTex** [comment]
@lukelabonte once more into the breach! d1b53ae6a2f3d2c03a454d1b6bf3a804af19d97c

**Review by lukelabonte — CHANGES_REQUESTED** [review]
Thanks, this addresses the main status issue I was worried about.

The new `hasActiveNonApprovalWork` shape looks right to me. A plain non-required `WAITING` check now falls through and counts as active work, and the new regression test covers the pending-vs-notStarted case I had in mind.

I do think there's one related UI classification issue still worth tightening up before merge. In `parseStatusChecks`, this still seems broader than the new status logic:

```swift
let isRequiredAggregateWork = !missingRequiredContexts.isEmpty
    && check.status?.uppercased() != "WAITING"
    && !isLegacyApprovalContext
```

Because all `WAITING` checks skip `isRequiredAggregateWork`, a plain non-required `WAITING` check like `build` can still end up as `isNonBlocking == true`. Then `nonBlockingCheckSummary` renders it as `1 waiting for approval`, even though it's active CI and not an approval gate.

Could we make the `isNonBlocking` path use the same approval-gate shape as `hasActiveNonApprovalWork`? I think the new regression test should also assert that the `build` check is not non-blocking, or that it doesn't show up in the non-blocking summary.

Requesting changes for that last bit, but I think this is close.

**Comment by SeanMcTex** [comment]
> Could we make the `isNonBlocking` path use the same approval-gate shape as `hasActiveNonApprovalWork`? I think the new regression test should also assert that the `build` check is not non-blocking, or that it doesn't show up in the non-blocking summary.

Good eye, sir. Addressed in 77726e8.

**Review by lukelabonte — APPROVED** [review]
